bin/add_geneticmodels_to_svvcf.py

```diff
@@ -0,0 +1,146 @@
+    ped, individuals = read_ped(args.pedigree_file)
+    if len(ped.keys()) < 2:
+        exit("Cannot add genetic models to single samples")
+    vcf_object = VariantFile(args.input_vcf)
+    vcf_object.header.info.add(
+        "GeneticModel",
+        ".",
+        "String",
+        "Genetic model for variant"
+    )
+    vcf_object.header.info.add(
+        "HOMHEM",
+        ".",
+        "String",
+        "If variant follows geneticmodel and is a homo- or hemi-zygous deletion"
+    )
+    vcf_out = VariantFile(args.out_vcf, "w", header=vcf_object.header)
+    vcf_out.close()
+    with open(args.out_vcf, "a") as vcf_out:
+        for var in vcf_object.fetch():
+            var_dict = cmdvcf.parse_variant(var,vcf_object.header)
+            genetic_model, is_del_hemi_homo = genetic_model_of_variant(var_dict,ped,individuals)
+            var_dict['INFO']['GeneticModel'] = genetic_model
+            if is_del_hemi_homo is not None:
+                var_dict['INFO']['HOMHEM'] = is_del_hemi_homo
+            vcf_str = cmdvcf.vcf_string(var_dict,vcf_object.header)
+            vcf_out.write(vcf_str)
+
+
+def genotype_sum_per_individual(gt:dict):
+    """
+    get sum of genotype, return dict with sample-id and sum
+    Hom alt = 2, hom ref = 0, het = 1
+    """
+    sample_gt_sum = {}
+    for sample in gt:
+        sample_id = sample.get('_sample_id')
+        genotype = sample.get('GT')
+        genotype = genotype.replace('.',"0")
+        gt_list = genotype.split("/")
+        gt_sum = sum(int(x) for x in gt_list)
+        sample_gt_sum[sample_id] = gt_sum
+    return sample_gt_sum
+
+def genetic_model_of_variant(var_dict:dict,ped:dict,individuals:dict):
+    """
+    Returns a genetic model that is compliant with disease
+    """
+    sample_gt_sum = genotype_sum_per_individual(var_dict['GT'])
+    chrom = var_dict['CHROM']
+    x = 0
+    if chrom == 'X':
+        x = 1
+    mother_gt   = sample_gt_sum[individuals['mother']]
+    mother_phe  = ped[individuals['mother']]['PHENO']
+    father_gt   = sample_gt_sum[individuals['father']]
+    father_phe  = ped[individuals['father']]['PHENO']
+    proband_gt  = sample_gt_sum[individuals['proband']]
+    proband_sex = ped[individuals['proband']]['SEX']
+
+    gm_search = f"{proband_sex}{proband_gt}{x}{father_gt}{father_phe}{mother_gt}{mother_phe}"
+    # ignore homozygous ref proband GTs. Is this a problem?
+    genetic_model = GM.get(gm_search,'NA')
+    # at the moment rediculous genetic patterns including homozygous dominant traits are marked *, compound finder cannot handle these
+    genetic_model = genetic_model.replace('*','')
+    hemi_homo_del = None
+    if var_dict['INFO']['SVTYPE'] == 'DEL':
+        hemi_homo_del = is_del_hemi_homo(proband_gt,proband_sex,x)
+    return genetic_model, hemi_homo_del
+    
+def is_del_hemi_homo(proband_gt,proband_sex,x):
+    """
+    checks if deletion is complete loss of genetic material
+    Boys on X special, ladies on X maybe not needed to check
```

Comment:
Could you give a mention of what the `proband_sex` values mean

Reply:
Or maybe even better. Why not put constants at the top of the module

```
SEX_MALE = 1
SEX_FEMALE = 2
```

and then

```
if proband_sex == SEX_MALE and proband_gt < 0:
```